modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/test/util/MockLiferayPortletRenderRequestUtil.java

```diff
@@ -0,0 +1,111 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.dashboard.web.internal.portlet.test.util;
+
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.portlet.bridges.mvc.constants.MVCRenderConstants;
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.test.portlet.MockLiferayPortletRenderRequest;
+import com.liferay.portal.kernel.test.portlet.MockLiferayPortletURL;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ProxyUtil;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.Locale;
+import java.util.Objects;
+
+import javax.portlet.PortletContext;
+import javax.portlet.PortletPreferences;
+import javax.portlet.PortletRequestDispatcher;
+
+/**
+ * @author David Arques
+ */
+public class MockLiferayPortletRenderRequestUtil {
+
+	public static MockLiferayPortletRenderRequest
+			getMockLiferayPortletRenderRequest(
+				String[] assetVocabularyNames, Company company, Locale locale)
+		throws Exception {
+
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			new MockLiferayPortletRenderRequest();
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			WebKeys.COMPANY_ID, company.getCompanyId());
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			StringBundler.concat(
+				mockLiferayPortletRenderRequest.getPortletName(), "-",
+				WebKeys.CURRENT_PORTLET_URL),
+			new MockLiferayPortletURL());
+
+		String path = "/view.jsp";
+
+		mockLiferayPortletRenderRequest.setParameter("mvcPath", path);
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			MVCRenderConstants.
+				PORTLET_CONTEXT_OVERRIDE_REQUEST_ATTIBUTE_NAME_PREFIX + path,
+			ProxyUtil.newProxyInstance(
+				PortletContext.class.getClassLoader(),
+				new Class<?>[] {PortletContext.class},
+				(PortletContextProxy, portletContextMethod,
+				 portletContextArgs) -> {
+
+					if (Objects.equals(
+							portletContextMethod.getName(),
+							"getRequestDispatcher") &&
+						Objects.equals(portletContextArgs[0], path)) {
+
+						return ProxyUtil.newProxyInstance(
+							PortletRequestDispatcher.class.getClassLoader(),
+							new Class<?>[] {PortletRequestDispatcher.class},
+							(portletRequestDispatcherProxy,
+							 portletRequestDispatcherMethod,
+							 portletRequestDispatcherArgs) -> null);
+					}
+
+					throw new UnsupportedOperationException();
+				}));
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _getThemeDisplay(company, locale));
+
+		PortletPreferences portletPreferences =
+			mockLiferayPortletRenderRequest.getPreferences();
+
+		portletPreferences.setValues(
+			"assetVocabularyNames", assetVocabularyNames);
+
+		return mockLiferayPortletRenderRequest;
+	}
+
+	private static ThemeDisplay _getThemeDisplay(Company company, Locale locale)
```

Comment:
Probably every class needs their own ThemeDisplay, some maybe need Locale, other not. Are you sure all of them needs the same parameters?

Reply:
This util class is meant just for these 2 classes, not for all the module. Maybe I should use another name

Reply:
If you use it in only two classes, better if you move the code there. Once we have a third class we will think about extract common code.

Reply:
https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
good point! And with a Fowler's quote, ok, I'm going to remove the last commit, I agree with you ;)